### PR TITLE
[CodeCompletion] NFC: rename `CallParameter*` to `CallArgument*`

### PR DESCRIPTION
**Pull request by rintaro**
Since these chunks are for call sites, the correct term is "argument".

(Subset of #38453)

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: include/swift/IDE/CodeCompletion.h [hunk 1]
```diff
@@ -140,27 +140,27 @@ class CodeCompletionStringChunk {
 
     /// The first chunk of a substring that describes the parameter for a
     /// function call.
-    CallParameterBegin,
```
Comment (benlangmuir) [inline include/swift/IDE/CodeCompletion.h:143]: We should update the doc comments for this and all the other cases too.

Comment (rintaro) [reply include/swift/IDE/CodeCompletion.h:143]: Updated the comments

**Review by benlangmuir — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**